src/components/views/auth/PasswordLogin.tsx

```diff
@@ -116,25 +116,25 @@ export default class PasswordLogin extends React.PureComponent<IProps, IState> {
     };
 
     private onUsernameChanged = (ev: React.ChangeEvent<HTMLInputElement>): void => {
-        this.props.onUsernameChanged(ev.target.value);
+        this.props.onUsernameChanged!(ev.target.value);
     };
 
     private onUsernameBlur = (ev: React.FocusEvent<HTMLInputElement>): void => {
-        this.props.onUsernameBlur(ev.target.value);
+        this.props.onUsernameBlur!(ev.target.value);
```

Comment:
Should there not better be a `?` used here? Per props it is optional.

```suggestion
        this.props.onUsernameBlur?.(ev.target.value);
```

(applies to other pro usages as well)

Reply:
`onUsernameBlur` is only ever used if `this.props.onUsernameBlur` is set, it'd be a lot cleaner as an inline anonymous function but we don't like those in class components, so the null-guard generated by TS is redundant